app/src/components/testCaseList/testCaseCard/testCaseCard.jsx

```diff
@@ -0,0 +1,127 @@
+import { messages } from './messages';
+import styles from './testCaseCard.scss';
+
+const cx = classNames.bind(styles);
+
+const StatusIcon = ({ status }) => {
+  const iconMap = {
+    passed: <div className={cx('priority-icon')}>{Parser(PriorityIcon)}</div>,
+    failed: '✗',
+    skipped: '○',
+    in_progress: '⟳',
+  };
+
+  return (
+    <div className={cx('status-icon', `status-icon--${status}`)}>{iconMap[status] || '?'}</div>
+  );
+};
+
+StatusIcon.propTypes = {
+  status: PropTypes.string.isRequired,
+};
+
+export const TestCaseCard = ({ testCase, onEdit, onDelete, onDuplicate, onMove }) => {
+  const { formatMessage } = useIntl();
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const menuItems = [
+    {
+      label: formatMessage(messages.duplicate),
+      onClick: () => onDuplicate(testCase),
+    },
+    {
+      label: formatMessage(messages.editTestCase),
+      onClick: () => onEdit(testCase),
+    },
+    {
+      label: formatMessage(messages.moveTestCaseTo),
+      onClick: () => onMove(testCase),
+    },
+    {
+      label: formatMessage(messages.deleteTestCase),
+      onClick: () => onDelete(testCase),
+      className: 'delete-menu-item',
+    },
+  ];
+
+  return (
+    <div className={cx('test-case-card')}>
+      <div className={cx('card-content')}>
+        <div className={cx('name-column')}>
+          <div className={cx('name-section')}>
+            <StatusIcon status={testCase.status} />
+            <div className={cx('name-content')}>
+              <div className={cx('test-name')} title={testCase.name}>
+                {testCase.name}
+              </div>
+              <div className={cx('tags-section')}>
+                <TagList tags={testCase.tags} isCustom />
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div className={cx('execution-column')}>
+          <div className={cx('execution-content')}>
+            <div className={cx('execution-time')}>{testCase.lastExecution}</div>
+            <div className={cx('menu-section')}>
+              <PopoverControl
+                items={menuItems}
+                placement="bottom-end"
+                isOpened={isMenuOpen}
+                setIsOpened={setIsMenuOpen}
+              >
+                <button className={cx('dots-menu-trigger')}>⋯</button>
+              </PopoverControl>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+TestCaseCard.propTypes = {
+  testCase: PropTypes.shape({
+    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+    name: PropTypes.string.isRequired,
+    status: PropTypes.string.isRequired,
+    tags: PropTypes.arrayOf(PropTypes.string),
+    lastExecution: PropTypes.string.isRequired,
+  }).isRequired,
+  onEdit: PropTypes.func,
+  onDelete: PropTypes.func,
+  onDuplicate: PropTypes.func,
+  onMove: PropTypes.func,
+};
+
+TestCaseCard.defaultProps = {
```

Comment:
Can you typescript based approach for new code? With it we won't need propTypes and default props